F2Py.cmake

```diff
@@ -69,11 +77,54 @@ function (f2py_add_module target_name)
   list(FILTER F2PY_ADD_MODULE_GEN_2 INCLUDE REGEX "${target_name}-f2pywrappers.*")
 
   set(F2PY_ADD_MODULE_LOG_FILE ${CMAKE_CURRENT_BINARY_DIR}/${target_name}.log)
+
+  set(F2PY_ADD_MODULE_INC)
+  if (F2PY_ADD_MODULE_INCLUDE_DIRS)
+    STRING(JOIN ":" _joined_dirs ${F2PY_ADD_MODULE_INCLUDE_DIRS})
+    set(F2PY_ADD_MODULE_INC --include-paths ${_joined_dirs})
+  endif()
   
   if (NOT F2PY_ADD_MODULE_PYF_FILE)
-    set(F2PY_ADD_MODULE_PYF_FILE ${CMAKE_CURRENT_BINARY_DIR}/${target_name}.pyf)
+    # Set directory name to output generated file *.pyf files
+    # *module.c and *-f2pywrappers.f
+    set(model_out ${CMAKE_CURRENT_BINARY_DIR}/f2py)
```

Comment:
Generally speaking, please be conservative when defining new variables. Adding variables that are trivial is making the code harder to read, especially when there is a long gap between the definition and the use. I think you can do very well without `model_out`, just insert the path verbatim where it is used later. There is no need to define an `f2py` subdirectory in the build path anyway.

Reply:
Do I understand correctly that the "established prefix" is `F2PY_ADD_MODULE`? If it is so, then it seems there is no problem with setting a variable without it (as with `model_out`), because a function in CMake creates a new scope and until we not use `set (models_out some_value PARENT_SCOPE)` it will not leak out. If `model_out` already exists somewhere re/setting in function's scope remove previous values.

Or are there more subtle issues which I haven't taken into account?

Reply:
I removed `model_out` variable and `f2py` directory. All generated pyf files are in ${CMAKE_CURRENT_BINARY_DIR}. 

So the above comment is related to other variables in f2py_add_module function.

Reply:
> Do I understand correctly that the "established prefix" is `F2PY_ADD_MODULE`?

Yes.

> If it is so, then it seems there is no problem with setting a variable without it (as with `model_out`), because a function in CMake creates a new scope and until we not use `set (models_out some_value PARENT_SCOPE)` it will not leak out.

Are you sure? Because I also read up on that again and my understanding was that functions do not have a local scope. Only add_subdirectory generates a new scope.

> If `model_out` already exists somewhere re/setting in function's scope remove previous values.

Yes, I am not concerned about that, because that's intended behavior. However, my understanding was that local variables in functions need to be prefixed with the function name to make collisions with existing variables unlikely.

Reply:
> Are you sure? 

`https://cmake.org/cmake/help/latest/command/function.html` says `A function opens a new scope: see set(var PARENT_SCOPE) for details.` 

I also checked minimal example in `https://levelup.gitconnected.com/cmake-variable-scope-f062833581b7`. It works as described.

Reply:
Sorry, I haven't read next comment.